cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -599,8 +648,22 @@ public QueryHandle execute(String statement) throws ExploreException, SQLExcepti
       // It looks like the username and password below is not used when security is disabled in Hive Server2.
       SessionHandle sessionHandle = cliService.openSession("", "", sessionConf);
       try {
+        // Switch database to the one being passed in.
+        final OperationHandle useDatabaseHandle = doExecute(sessionHandle, String.format("USE %s", namespace));
```

Comment:
Will see if that works here.

Yes, the error message is meaningful. However, we do want to do validation of namespaces but that will not happen in this PR. Theres some work to be done around that bit in general for the whole of CDAP.